dashboard/app/controllers/followers_controller.rb

```diff
@@ -25,14 +24,23 @@ def student_register
       return render 'student_user_new', formats: [:html]
     end
 
-    Retryable.retryable on: [Mysql2::Error, ActiveRecord::RecordNotUnique], matching: /Duplicate entry/ do
-      if @user.save && @section&.add_student(@user)
-        sign_in(:user, @user)
-        redirect_to root_path, notice: I18n.t('follower.registered', section_name: @section.name)
-        return
+    if @user && @user.display_captcha? && !verify_recaptcha
```

Comment:
Mostly curious: It looks like all of the other error redirects in this controller are done in load_section.  Is there a reason you put the captcha logic here instead of load_section?  (I'm not sure which is better, just curious around your thinking.)

Reply:
My thought process was that the `load_section` method is responsible for finding the section and validating that the user is allowed to join the section. If any of those checks fail, we increment the section-attempts hash and allow `load_section` to redirect as it did before. Checking for captcha verification seemed like a check that would come after loading the section... Not sure if that logic checks out. I'm open to suggestions!